os/mm/mm_heap/mm_initialize.c

```diff
@@ -127,6 +127,21 @@ void mm_addregion(FAR struct mm_heap_s *heap, FAR void *heapstart, size_t heapsi
 
 	heapbase = MM_ALIGN_UP((uintptr_t)heapstart);
 	heapend  = MM_ALIGN_DOWN((uintptr_t)heapstart + (uintptr_t)heapsize);
+	if ((heapbase < (uintptr_t)heapstart)) {
+		/* heapbase cannot be smaller than heapstart.
+		 * If this happens, heapbase can be overflowed from alignment.
+		 */
+		return;
+	}
+
+	if (heapbase >= (uintptr_t)heapend) {
+		/* heapbase is aligned up from heapstart,
+		 * and heapend is the summation of heapstart and heapsize.
+		 * If heapsize is very small, align-up address can be equal to or greater than heapend.
+		 * In this case, heapbase cannot be the base address of heap.
```

Comment:
This line has little bit different meaning against the statement.
This statement means there is no meaningful `size` for new heap region.
But this line says wrong `base address` of heap.